irohad/ametsuchi/impl/postgres_wsv_query.cpp

```diff
@@ -19,266 +19,193 @@
 
 namespace iroha {
   namespace ametsuchi {
-
-    using std::string;
-
-    using nonstd::optional;
-    using nonstd::nullopt;
-    using model::Account;
-    using model::Asset;
-    using model::AccountAsset;
-    using model::Peer;
-    using model::Domain;
-
     PostgresWsvQuery::PostgresWsvQuery(pqxx::nontransaction &transaction)
-        : transaction_(transaction), log_(logger::log("PostgresWsvQuery")) {}
+        : transaction_(transaction), log_(logger::log("PostgresWsvQuery")), execute_{makeExecute(transaction_, log_)} {
+    }
 
     bool PostgresWsvQuery::hasAccountGrantablePermission(
         const std::string &permitee_account_id,
         const std::string &account_id,
         const std::string &permission_id) {
-      pqxx::result result;
-      try {
-        result = transaction_.exec(
-            "SELECT * FROM account_has_grantable_permissions WHERE "
-            "permittee_account_id = "
-            + transaction_.quote(permitee_account_id)
-            + " AND account_id = "
-            + transaction_.quote(account_id)
-            + " AND permission_id = "
-            + transaction_.quote(permission_id)
-            + ";");
-      } catch (const std::exception &e) {
-        log_->error(e.what());
-        return false;
-      }
-      return result.size() == 1;
+      return execute_(
+                 "SELECT * FROM account_has_grantable_permissions WHERE "
+                 "permittee_account_id = "
+                 + transaction_.quote(permitee_account_id)
+                 + " AND account_id = " + transaction_.quote(account_id)
+                 + " AND permission_id = " + transaction_.quote(permission_id)
+                 + ";")
```

Comment:
It would look much better with fmt instead of weird concatenatios :(

Reply:
Please refer to https://soramitsu.atlassian.net/browse/IR-806